src/pobox.erl

```diff
@@ -127,22 +207,54 @@ post(Box, Msg) ->
 %%      with the keep_old buffer type because it tells you the message will
 %%      be dropped.
 -spec post_sync(name(), term(), timeout()) -> ok | full.
-post_sync(Box, Msg, Timeout) ->
+post_sync(Box, Msg, Timeout) when ?PROCESS_NAME_GUARD(Box), is_integer(Timeout), Timeout >= 0 ->
```

Comment:
Should I bother with the guards? I guess it would produce a shorter stack trace but gets executed all the time so I am not sure.